pygmt/src/sphdistance.py

```diff
@@ -22,7 +22,7 @@
     V="verbose",
 )
 @kwargs_to_strings(I="sequence", R="sequence")
-def sphdistance(table, **kwargs):
+def sphdistance(x=None, y=None, z=None, data=None, **kwargs):
```

Comment:
As discussed in #1479, the order should be `data, x, y, z`:
```suggestion
def sphdistance(data=None, x=None, y=None, z=None, **kwargs):
```